exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/statistics/StatisticsMergeBatch.java

```diff
@@ -0,0 +1,532 @@
+    dataSrcVecMap = Maps.newHashMap();
+    copySrcVecMap = Maps.newHashMap();
+    MajorType mt = null;
+
+    ErrorCollector collector = new ErrorCollectorImpl();
+    GregorianCalendar calendar = new GregorianCalendar(TimeZone.getTimeZone("UTC"));
+
+    calendar.setTimeInMillis(System.currentTimeMillis());
+    createKeyColumn("schema", ValueExpressions.getBigInt(schema++), copySrcVecMap);
+    createKeyColumn("computed", ValueExpressions.getDate(calendar), copySrcVecMap);
+
+    for (VectorWrapper<?> vw : incoming) {
+      addVectorToOutgoingContainer(vw.getField().getLastName(), vw, collector);
+      /*MaterializedField ds = vw.getField();
+      String field = vw.getField().getLastName();
+      // Input map vector
+      MapVector mapVector = (MapVector) vw.getValueVector();
+      assert mapVector.getPrimitiveVectors().size() > 0;
+      // Proceed to create output map vector with same name e.g. statcount etc.
+      mt = mapVector.getField().getType();
+      MaterializedField mf = MaterializedField.create(functions.get(field), mt);
+      assert !dataSrcVecMap.containsKey(mf);
+      ValueVector vector = TypeHelper.getNewVector(mf, oContext.getAllocator());
+      container.add(vector);
+      MapVector outputMapVector = (MapVector) vector;
+
+      for (ValueVector vv : mapVector) {
+        String fieldName = vv.getField().getLastName();
+        if (!keyList.contains(fieldName)) {
+          throw new UnsupportedOperationException("Unpivot data vector " +
+              ds + " contains key " + fieldName + " not contained in key source!");
+        }
+        if (vv.getField().getType().getMinorType() == TypeProtos.MinorType.MAP) {
+          throw new UnsupportedOperationException("Unpivot of nested map is not supported!");
+        }
+        if (field.equals("column")) {
+          outputMapVector.addOrGet(fieldName, vv.getField().getType(), vv.getClass());
+        } else {
+          List<LogicalExpression> args = Lists.newArrayList();
+          //TODO: Something else to access value of col such as emp_id?
+          args.add(SchemaPath.getSimplePath(vv.getField().getPath()));
+          //TODO: Put in the mapVector
+          LogicalExpression call = FunctionCallFactory.createExpression(functions.get(field), args);
+          //TODO: Is this sufficient to add to new Map?
+          ValueVector vector1 = addMapVector(fieldName, outputMapVector, call);
+          if (collector.hasErrors()) {
+            throw new SchemaChangeException("Failure while materializing expression. "
+                + collector.toErrorString());
+          }
+        }
+      }
+      dataSrcVecMap.put(ds, outputMapVector);*/
+    }
+    //Now create NDV in the outgoing container which was not avaliable in the incoming
+    for (VectorWrapper<?> vw : incoming) {
+      if (vw.getField().getLastName().equalsIgnoreCase("sum_width")) {//NullableFloat8 type
+        addVectorToOutgoingContainer("ndv", vw, collector);
```

Comment:
Again, we are hard-coding knowledge of certain functions here in the code, makes it hard to extend the set of functions.

See comment below about using hard-coded strings rather than declared constants.